src/main/java/appeng/core/Registration.java

```diff
@@ -150,18 +143,11 @@ void preInitialize( final FMLPreInitializationEvent event )
 
 		MinecraftForge.EVENT_BUS.register( OreDictionaryHandler.INSTANCE );
 
-		final ApiDefinitions definitions = api.definitions();
+		ApiDefinitions definitions = api.definitions();
```

Comment:
If possible yes. It has a few advantage like explicitly stating they will never change or actually preventing wrongful reassignments.

Also `javac` can perform a more optimisations. Consider something like this
```
final int a = 42;
final int b = 24;
final int c = a - b:
```
In this case the compiler will replace `a - b` with `18` at compile time. Without every variable being final it will keep `a - b` and calculate it at runtime. With some luck the JIT might optimise it away. But so it will be guaranteed. It might not be much, but if it is called a few thousand times per frame it can add up.